### ISSUE TITLE
UUID in ClosureEntity

### ISSUE THREAD

**Issue by amirsaber** [typeorm/typeorm#1089]
Thanks for your amazing library. I am trying to adopt our app to this library, I am facing an issue when I use uuid in a closure entity. If I set integer everything works well.
Here is my code:
```js
@ClosureEntity()
export class Group extends BaseEntity {

  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({
    nullable: false
  })
  name: string;

  @TreeChildren()
  children: Group;

  @TreeParent()
  parent: Group;

  @TreeLevelColumn()
  level: number;
}
```
Here is the error:
```
{ QueryFailedError: syntax error at or near "-"
    at new QueryFailedError (/media/amirsaber/ExtraDrive1/Playground/access/node_modules/typeorm/error/QueryFailedError.js:27:28)
    at Query.callback (/media/amirsaber/ExtraDrive1/Playground/access/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:216:38)
    at Query.handleError (/media/amirsaber/ExtraDrive1/Playground/access/node_modules/pg/lib/query.js:143:17)
    at Connection.connectedErrorHandler (/media/amirsaber/ExtraDrive1/Playground/access/node_modules/pg/lib/client.js:132:26)
    at emitOne (events.js:115:13)
    at Connection.emit (events.js:210:7)
    at Socket.<anonymous> (/media/amirsaber/ExtraDrive1/Playground/access/node_modules/pg/lib/connection.js:118:12)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at addChunk (_stream_readable.js:264:12)
  name: 'QueryFailedError',
  length: 90,
  severity: 'ERROR',
  code: '42601',
  detail: undefined,
  hint: undefined,
  position: '91',
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'scan.l',
  line: '1086',
  routine: 'scanner_yyerror',
  query: 'INSERT INTO "group_closure"("ancestor", "descendant", "level") SELECT "ancestor", 6bdb2dd3-cdd4-4305-88c6-e9d2a8234bca, "level" + 1 FROM "group_closure" WHERE "descendant" = 0 UNION ALL SELECT 6bdb2dd3-cdd4-4305-88c6-e9d2a8234bca, 6bdb2dd3-cdd4-4305-88c6-e9d2a8234bca, 1',
  parameters: [] }
```
Am I doing anything wrong?

**Comment by pleerock** [comment]
Closure entities are subject to refactoring planned on 0.3.0. Its simply not supports uuid right now. You can wait for 0.3.0 or try to fix it by yourself and send a PR. Or at least send a failing test which I can fix quickly. Looks like fix for this issue is to escape "6bdb2dd3-cdd4-4305-88c6-e9d2a8234bca" in SELECT statement (see failing query)

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.
